src/utilities/extendSchema.js

```diff
@@ -150,8 +151,30 @@ export function extendSchema(
         }
         directiveDefinitions.push(def);
         break;
-      case Kind.SCALAR_TYPE_EXTENSION:
       case Kind.INTERFACE_TYPE_EXTENSION:
+        const extendedInterfaceTypeName = def.name.value;
+        const existingInterfaceType = schema.getType(extendedInterfaceTypeName);
+        if (!existingInterfaceType) {
+          throw new GraphQLError(
+            `Cannot extend interface "${extendedInterfaceTypeName}" because ` +
+              'it does not exist in the existing schema.',
+            [def],
+          );
```

Comment:
You need to test it similar to how it's down for object type:
https://github.com/graphql/graphql-js/blob/b333b30b238d040d6a8cf959d54f037ee26ae2c6/src/utilities/__tests__/extendSchema-test.js#L985